src/libxtp/gaussian_quadrature.cc

```diff
@@ -0,0 +1,225 @@
+/*
+ *            Copyright 2009-2019 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <votca/tools/constants.h>
+#include <votca/xtp/gauss_hermite_quadrature_constants.h>
+#include <votca/xtp/gauss_laguerre_quadrature_constants.h>
+#include <votca/xtp/gauss_legendre_quadrature_constants.h>
+#include <votca/xtp/gaussian_quadrature.h>
+#include <votca/xtp/threecenter.h>
+
+namespace votca {
+namespace xtp {
+
+// Constructor
+GaussianQuadrature::GaussianQuadrature(const Eigen::VectorXd& energies,
+                                       const TCMatrix_gwbse& Mmn)
+    : _energies(energies), _Mmn(Mmn) {}
+
+void GaussianQuadrature::configure(options opt, const RPA& rpa) {
+  _opt = opt;
+  if (_opt.quadrature_scheme == "laguerre") {
+    Gauss_Laguerre_Quadrature_Constants glqc;
+    _quadpoints = glqc.getPoints(_opt.order);
+    _quadadaptedweights = glqc.getAdaptedWeights(_opt.order);
+    CalcDielInvVector(rpa);
+  } else if (_opt.quadrature_scheme == "legendre") {
+    Gauss_Legendre_Quadrature_Constants glqc;
+    _quadpoints = glqc.getPoints(_opt.order);
+    _quadadaptedweights = glqc.getAdaptedWeights(_opt.order);
+    CalcDielInvVector(rpa);
+  } else if (_opt.quadrature_scheme == "modified_legendre") {
+    Gauss_Legendre_Quadrature_Constants glqc;
+    _quadpoints = glqc.getPoints(_opt.order);
+    _quadadaptedweights = glqc.getAdaptedWeights(_opt.order);
+    CalcDielInvVector(rpa);
+  } else if (_opt.quadrature_scheme == "hermite") {
+    Gauss_Hermite_Quadrature_Constants glqc;
+    _quadpoints = glqc.getPoints(_opt.order);
+    _quadadaptedweights = glqc.getAdaptedWeights(_opt.order);
+    CalcDielInvVector(rpa);
+  } else {
+    std::cout << "There no such a thing as the integration scheme you asked"
+              << std::endl;
+  }
+}
+
+// This function calculates and stores inverses of the microscopic dielectric
+// matrix in a matrix vector
+void GaussianQuadrature::CalcDielInvVector(const RPA& rpa) {
+  _dielinv_matrices_r.resize(_opt.order);
+  double halfpi = std::acos(0.0);
+#pragma openmp parallel schedule(guided)
+  for (Index j = 0; j < _opt.order; j++) {
+    if (_opt.quadrature_scheme == "legendre") {
+      double newpoint = std::tan(halfpi * _quadpoints(j));
+      Eigen::MatrixXcd eps_inv_j =
+          rpa.calculate_epsilon_complex(0.0, newpoint).inverse();
```

Comment:
You calculate it on the real axis, do you think we could use on of the other RPA methods here?

Reply:
This is on the imaginary axis. 
I built this function because in the future we might need a more flexible function able to accept complex number (both real and imaginary part different from zero)

Reply:
Do you think in this way I slowed things down?

Reply:
On the imaginary axis the result is purely real/imagninary (I forgot which) so if you use a complex type there, it is 4x more expensive at least.

Reply:
I changed both in gaussian_quadrature and sigma_cda so that the dielectric function is always real (hence inverting real matrices) even when complex argument are given